components/core/src/metrics.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2016 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::thread;
+use std::sync::mpsc::{channel, sync_channel, Sender, Receiver, SyncSender};
+use std::sync::{Once, ONCE_INIT};
+use statsd::Client;
+use env;
+
+// Statsd Application name
+pub const APP_NAME: &'static str = "bldr";
```

Comment:
If statsd is the wrong strategy for hap-sup, why is it the right strategy for the build service? #1469 uses the prometheus style for distribution, and includes an example of how some internal supervisor metrics could be exposed, but doesn't provide any sort of contract for a service to expose metrics which are then collected. As far as protocols to do that go, statsd is pretty great, and there is an existing prometheus bridge. Whatever we decide for the sup is eventually what we should use for builder, as these are all open components into a broader ecosystem. The other direction is a fine way (we add instrumentation to the builder services, which grows into the supervisor) - but that's how it should work, because relaying application instrumentation is part of what the supervisor needs to do long term.